web/data/data.go

```diff
@@ -398,6 +398,62 @@ func normalDocs(c echo.Context) error {
 	return c.JSON(http.StatusOK, res)
 }
 
+func getDesignDoc(c echo.Context) error {
+	doctype := c.Get("doctype").(string)
+	ddoc := c.Param("designdocid")
+
+	if err := middlewares.AllowWholeType(c, permission.GET, doctype); err != nil {
+		return err
+	}
+	return proxy(c, "_design/"+ddoc)
+}
+
+func getDesignDocs(c echo.Context) error {
+	doctype := c.Get("doctype").(string)
+	if err := permission.CheckReadable(doctype); err != nil {
+		return err
+	}
+	if err := middlewares.AllowWholeType(c, permission.GET, doctype); err != nil {
+		return err
+	}
+	return proxy(c, "_design_docs")
+}
+
+func copyDesignDoc(c echo.Context) error {
+	doctype := c.Param("doctype")
+	ddoc := c.Param("designdocid")
+
+	c.Request().Method = "COPY"
```

Comment:
It's quite ugly to have to modify the incoming request :-/

Reply:
Indeed, but what else could we do as the COPY is not supported by go, and with no possible override from echo? :/